src/drivers/intel/cavs/hda-dma.c

```diff
@@ -323,6 +332,35 @@ static int hda_dma_link_copy_ch(struct dma_chan_data *chan, int bytes)
 	return ret;
 }
 
+static void hda_dma_host_start(struct dma_chan_data *channel)
+{
+	struct hda_chan_data *hda_chan = dma_chan_get_data(channel);
+
+	/* Force Host DMA to exit L1 only on start*/
+	if (!(hda_chan->state & HDA_STATE_RELEASE))
+		pm_runtime_put(PM_RUNTIME_HOST_DMA_L1, 0);
+
+	if (!hda_chan->irq_disabled)
+		return;
+
+	/* Register common L1 exit for all channels */
+	notifier_register(NULL, scheduler_get_data(SOF_SCHEDULE_LL_TIMER),
```

Comment:
Errors returned from Notifier API (like -ENOMEM are not handled).

Reply:
Done.